app/controllers/user_projects_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UserProjectsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_project
+
+  def assign
```

Comment:
Use "scaffold" names when possible so `create` and `destroy` instead of `assign` and `leave`